File Submission

Katijino opened the pull request:
File Submission

This is a pull request about improvement on settings page. This request includes file submission, retrieval, and proper storage using IndexedDB database. Along with a new layout and form fields for Education, Work Experience, Skills, and Interests.

## Type of change

- [ ] Bug fix (change which fixes an issue)
- [x] Enhancement (change which alters existing functionality)
- [ ] New feature (change which adds new functionality)
- [ ] This change requires a wiki/documentation update

## How Has This Been Tested?

- Ran tests with different pdf files, 3 of the same pdf files, saving 3 at once, 1 at a time, and ensured ID and file type was correct for submissions.
- Tested various inputs for form fields to see if they would be stored correctly in chrome local storage sync
- Attempted to break the dynamic fields by removing and adding fields while submitting

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas (definitely not going to do this lmao)
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

Katijino committed:
All recent commits & File saving feature

first commit to new branch, contains all changes from main contained in old SettingsBranch and the File Submission feature to actively save files to IndexedDB.

jimlo7760 committed:
fix the bug of upload file without clicking the submit button

jimlo7760 committed:
Apply the 10 MB file limitation to all file upload inputs.

anthonyfabius committed:
Updating Bootstrap and Button Variants

Katijino committed:
Fixed all known issues with File Submission

File submission was running before, but wasn't able to deal with only saving 3 files at once and saving an ID. New code added to remove files and added ID's corresponding to the different files being saved.

Katijino committed:
Commented Code

Commented code for easier understanding of the functions, for anyone trying to understand how to connect IndexedDB to their code.

Katijino committed:
Prettified Code!

Automated with github actions

anthonyfabius commented:
I don't think you got the file replacing feature down. Whenever I submit one file then replace it the past file is still there. Same goes for clearing. Clearing should remove all entries from the IndexedDB.

Submitting File1 as resume
![file1](https://github.com/EasyApp-RPI/EasyApp/assets/67553651/0b7b96e6-f432-4d6b-84f6-38eaae29eac1)

After clicking clear and submitting File2 as resume
![file2](https://github.com/EasyApp-RPI/EasyApp/assets/67553651/651b64a8-4bc4-4923-97cc-7c59b02ba07c)

jimlo7760 committed:
Create the education component file

jimlo7760 committed:
50% done of the education component

Still have to deal with the different fields and layout

Katijino committed:
Names part of Settings

This commit is to Implement the Names part of the Settings page. There is a bug where the saved information isn't being accessed in the main App.tsx in HandleSubmit, as it is saved in the side file. This must be fixed

anthonyfabius committed:
Config and Dependencies

anthonyfabius committed:
Files & Personal Comp Implementation

I added the files for the expected number of components we are going to have. I also began working on the personal information component. I got it working so the rest should really just be a copy and paste

anthonyfabius committed:
Finished Personal Component

Saves and updates to local storage. Is gonna take a little bit to do this for all the other fields but c'est la vie

anthonyfabius committed:
Working Interests and SkillsComp

The amount of blood sweat and tears I put into this commit is not funny. Thought I was gonna have to switch majors if this shit didn't work ngl.

anthonyfabius committed:
Working Education and Work Experience

More blood sweat and tears but a little less time :)

anthonyfabius committed:
Darkmode

Commit:
Prettified Code!

Automated with github actions

anthonyfabius commented:
@LeadFreeCandy review this shit, i actually blew my whole wednesday working on this. I even rebased just for you

anthonyfabius committed:
Half working clear button

Commit:
Prettified Code!

Automated with github actions
